set.mm

```diff
@@ -369106,6 +369106,95 @@ varying definitions (some start from 0, others start from 1), but
        renamer theorem like ~ cbval . The section
        ~ mmtheorems32.html#mm3146s also describes the
        metatheorem that underlies this.
+       </p>
+
+       <p>
+       Standard Metamath verifiers do not distinguish between axioms and
+       definitions (both are $a statements).
+       In practice, we require that definitions (1) not be creative
+       (a definition should not allow an expression
+       that previously qualified as a wff but was not provable
+       to become provable) and be eliminable
+       (there should exist an algorithmic method for converting any
+       expression using the definition into
+       a logically equivalent expression that previously qualified as a wff).
+       To ensure this, we have additional rules on almost all definitions
+       ($a statements with a label that does not begin with ax-).
+       These additional rules are not applied in a few cases where they
+       are too strict ( ~ df-bi , ~ df-clab , ~ df-cleq , and ~ df-clel );
+       see those definitions for more information.
+       These additional rules for definitions are checked by at least
+       mmj2's definition check (see
+       <A HREF="https://github.com/digama0/mmj2/blob/master/"
+       >mmj2 master</a> file mmj2jar/macros/definitionCheck.js).
+       This definition check relies on the database being very much like
+       set.mm, down to the names of certain constants and types, so it
+       cannot apply to all Metamath databases... but it is useful in set.mm.
+       In this definition check, a $a-statement with a given label and
+       typecode ` |- ` passes the test if and only if it
+       respects the following rules (these rules imply that we have
+       an unambiguous tree parse, which is also checked):
```

Comment:
These rules don't ensure an unambiguous parse; rather, they presume that the expression has a unique parse, and unambiguity and parse existence is checked separately.

Reply:
You're absolutely right, will fix.  What I meant != what I said :-).